### DISCUSSION TITLE
0.26.0

### DISCUSSION THREAD

**Discussion by flemming-n-larsen** [robocode-dev/tank-royale#110] [Announcements]
#### 🚀 Improvements

- GUI:
    - Replaced the `Server` -> `Select Server` with `Config` -> `Server Config` which makes it possible choose between
      using a local or remote server. In addition, it is possible to add, edit, and remove remote servers, and it is
      possible to specify a controller and a bot secret to allow the GUI and bots to access the remote server.
    - If a booted bot process exits, it is now removed from the active New Battle window.
- Booter:
    - When a pid of a bot process cannot be found, the Booter writes out a `load {pid}` to indicate that a process could
      not be found with the provided pid.

#### 🐞 Bug Fixes

- GUI
    - Control Panel was shown when server was stopped or rebooted. 

## 📄 Documentation

You can read the Robocode Tank Royale [documentation here](https://robocode-dev.github.io/tank-royale/index.html).

### 🔨 Try it out

Please head over to [My First Bot tutorial](https://robocode-dev.github.io/tank-royale/tutorial/my-first-bot.html) to learn how to set up your first bot for Robocode Tank Royale.

## ▶ Running Robocode

You need Java 11 as a minimum or newer, e.g. the newest version of Java available.

You can read the [installation guide] to get more details about installing both Java and Robocode.

The main application is the [GUI Application] which is a Java application.
You can read more about how to use the GUI Application [here](https://robocode-dev.github.io/tank-royale/articles/gui.html#gui-application).

The Robocode [GUI application] is run from the command line (shell or command prompt) in order to start and view
battles:

```shell
java -jar robocode-tankroyale-gui-0.26.0.jar
```

If you are new to Robocode, you need to download some bots and extract those to directories on your system.
These bot directories can be added from the menu of the GUI: `Config → Bot Root Directories`

## 🤖 Sample bots

These sample bots are currently available:

| Platform | Archive                            | Requirements                      |
|----------|------------------------------------|-----------------------------------|
| C#       | [sample-bots-csharp-0.26.0.zip] | Microsoft [.Net SDK] 6.0 or newer |
| Java     | [sample-bots-java-0.26.0.zip]   | Any [Java SDK] 11 or newer        |

All bots are put in zip archives, which should be installed in independent directories.
Each zip archive contains a ReadMe.md file with more information for the specific platform.

## 📦 Bot API

In order to start developing bots for Robocode, the following APIs are available.

#### 📦 Java:

Available as:

- Jar file: [robocode-tankroyale-bot-api-0.26.0.jar]
- Artifact at [Nexus Repository](https://s01.oss.sonatype.org/index.html#view-repositories;releases~browsestorage~/dev/robocode/tankroyale/robocode-tankroyale-bot-api/0.26.0/robocode-tankroyale-bot-api-0.26.0.jar)
- Artifact at [Maven Central Repository](https://search.maven.org/search?q=g:dev.robocode.tankroyale) (available after some time)

#### 📦 .Net:

Available as:
 - Artifact at [Nuget repository](https://www.nuget.org/packages/Robocode.TankRoyale.BotApi/0.26.0)

[My First Bot tutorial]: https://robocode-dev.github.io/tank-royale/tutorial/my-first-bot.html

[Installation guide]: https://robocode-dev.github.io/tank-royale/articles/installation.html

[GUI Application]: https://github.com/robocode-dev/tank-royale/releases/download/v0.26.0/robocode-tankroyale-gui-0.26.0.jar

[.Net SDK]: https://dotnet.microsoft.com/en-us/download/dotnet

[Java SDK]: https://robocode-dev.github.io/tank-royale/articles/installation.html#java-11-or-newer

[sample-bots-csharp-0.26.0.zip]: https://github.com/robocode-dev/tank-royale/releases/download/v0.26.0/sample-bots-csharp-0.26.0.zip

[sample-bots-java-0.26.0.zip]: https://github.com/robocode-dev/tank-royale/releases/download/v0.26.0/sample-bots-java-0.26.0.zip

[robocode-tankroyale-bot-api-0.26.0.jar]: https://s01.oss.sonatype.org/service/local/repositories/releases/content/dev/robocode/tankroyale/robocode-tankroyale-bot-api/0.26.0/robocode-tankroyale-bot-api-0.26.0.jar

<hr /><em>This discussion was created from the release <a href='https://github.com/robocode-dev/tank-royale/releases/tag/v0.26.0'>0.26.0</a>.</em>

**Comment by flemming-n-larsen** [comment]
I just updated the GUI app ([robocode-tankroyale-gui-0.26.0.jar](https://github.com/robocode-dev/tank-royale/releases/download/v0.26.0/robocode-tankroyale-gui-0.26.0.jar)) where a annoying bug was introduced where only a single instance of a booted bot could be stopped, but the other instances of the bot with the same name would continue running, and could not be stopped.

So I you downloaded this jar file before this comment, then please download the updated version [here](https://github.com/robocode-dev/tank-royale/releases/download/v0.26.0/robocode-tankroyale-gui-0.26.0.jar).

**Comment by SirStone** [comment]
This version GUI is missing the robocode-tankroyale-booter.jar

The GUI is spitting out errors about not finding the file
![Schermata del 2024-10-26 14-07-52](https://github.com/user-attachments/assets/339351db-5fd4-465d-8a97-f52401efdf49)

In fact the file is not present in the jar, while is present in 0.25.0 version.
![Schermata del 2024-10-26 14-06-00](https://github.com/user-attachments/assets/141a9148-790f-46ba-82f9-f0a6989e8d88)

**Reply by flemming-n-larsen** [reply]
@SirStone  Thank you for the notification. 😅
I have now uploaded a version 0.26.0 containing the missing Booter.

**Comment by SirStone** [comment]
@flemming-n-larsen Now is working again thanks.
